onadata/apps/api/viewsets/attachment_viewset.py

```diff
@@ -94,14 +95,16 @@ def count(self, request, *args, **kwargs):
         return Response(data=data)
 
     def list(self, request, *args, **kwargs):
-        if request.user.is_anonymous:
-            xform = request.query_params.get('xform')
-            if xform:
-                xform = XForm.objects.get(id=xform)
-                if not xform.shared_data:
-                    raise Http404(_("Not Found"))
+        xform = request.query_params.get('xform')
+        if xform:
+            xform = XForm.objects.get(id=xform)
```

Comment:
@DavisRayM Should we use a try and catch over here?

Reply:
Updated in the latest commit; Now raising a 404 if the XForm is not found